glean-core/rlb/src/test.rs

```diff
@@ -102,12 +156,16 @@ fn initialize_must_not_crash_if_data_dir_is_messed_up() {
         max_events: None,
         delay_ping_lifetime_io: false,
         channel: Some("testing".into()),
+        server_endpoint: Some("invalid-test-host".into()),
+        uploader: None
     };
 
-    initialize(cfg, ClientInfoMetrics::unknown());
-
-    // TODO: is this test working? is it waiting for init to finish?
-    dispatcher::block_on_queue();
+    reset_glean(cfg, ClientInfoMetrics::unknown(), false);
+    // Glean init is async and, for this test, it bails out early due to
```

Comment:
This is really ugly. We either need to make the dispatcher behave like the other RLB or introduce some new test-only function to track the progress of the async init. @badboy what do you think?

I'm worried about the dispatchers diverging in semantics, because other discrepancies may be lurking under the hood.

Reply:
As discussed making them more equal again seems preferrable.
This can be done in a followup, so please add a TODO and file a bug.

Reply:
Ah, the bug is at the end. I'd say add it above:
```suggestion
    // TODO(bug 1675215): ensure initialize runs through dispatcher.
    // Glean init is async and, for this test, it bails out early due to
```